ppa/archive/management/commands/hathi_rsync.py

```diff
@@ -26,22 +30,64 @@ def handle(self, *args, **kwargs):
+        # by default, sync data for all non-suppressed hathi source ids
+        digworks = DigitizedWork.objects.filter(
+            status=DigitizedWork.PUBLIC, source=DigitizedWork.HATHI
+        )
 
-        # NOTE: if htid is specified, should we verify that it's
-        # in the db and not suppressed? (should import first if not)
+        # if htids are specified via parameter, use them to filter
+        # the queryset, to ensure we only sync records that are
+        # in the database and not suppressed
+        if htids:
+            digworks = digworks.filter(source_id__in=htids)
+        # NOTE: report here on any skipped ids?
 
-        self.stdout.write(
-            self.style.SUCCESS("Synchronizing data for %d records" % len(htids))
-        )
-        # even if verbosity is zero we want an output file
+        # generate a list of unique source ids from the queryset
+        hathi_ids = digworks.values_list("source_id", flat=True).distinct()
+        self.stdout.write("Synchronizing data for %d records" % len(hathi_ids))
+        # we always want itemized rsync output, so we can report
+        # on which volumes were updated
         htimporter = HathiImporter(
-            source_ids=htids, rsync_output=self.verbosity or True
+            source_ids=hathi_ids, rsync_output=True, output_dir="/tmp"
         )
         logfile = htimporter.rsync_data()
-        self.stdout.write(self.style.SUCCESS("rsync output is in %s" % logfile))
+
+        # read the rsync itemized output to identify records where file
+        # sizes changed
+        updated_ids = set()
+        with open(logfile) as rsync_output:
+            for line in rsync_output:
+                # if a line indicates that a file was updated due
+                # to a change in size, use the path to determine the hathi id
+                if " >f.s" in line:
+                    # rsync itemized output is white-space delimited;
+                    # last element is the filename that was updated
+                    filename = line.rsplit()[-1].strip()
+                    # we only care about zip files and mets.xml files
+                    if not filename.endswith(".zip") and not filename.endswith(".xml"):
+                        continue
+                    # reconstruct the hathi id from the filepath
+                    ht_prefix, pairtree_dir = filename.split("/pairtree_root/")
+                    # get the directory one level up from the updated file
+                    pairtree_id = os.path.dirname(os.path.dirname(pairtree_dir))
+                    # use pairtree to determine the id based on the path
+                    # (handles special characters like those used in ARKs)
+                    htid = f"{ht_prefix}.{path2id(pairtree_id)}"
+                    updated_ids.add(htid)
+
+        # should this behavior only be when updating all?
+        # if specific htids are specified on the command line, maybe report on them only?
```

Comment:
It seems like it'd always be useful to know *which* files were updated by rsync, since it's not guaranteed that any are.